### PR TITLE
Added file bind extension methods

### PR DESCRIPTION
**Pull request by JoeStead**
We were getting quite a few extension method classes, moved them all into a folder as well

### CODE REVIEW

File: src/Extensions/BindExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+                }
+            }
+
+            return postedFiles;
+        }
+        
+        public static async Task BindAndSaveFiles(this HttpRequest request, string saveLocation)
+        {
+            
+            if (request.HasFormContentType)
```
Comment (jchannon) [inline src/Extensions/BindExtensions.cs]: This can call BindFiles to get the file list

File: src/Extensions/BindExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                    }
+
+                    postedFiles.Add(file);
+                }
+            }
+
+            return postedFiles;
+        }
+        
+        public static async Task BindAndSaveFiles(this HttpRequest request, string saveLocation)
```
Comment (jchannon) [inline src/Extensions/BindExtensions.cs]: I think we need a `BindFile` and `BindAndSaveFile` maybe? In MVC you can inject `IFormFile` or `IEnumerable<IFormFile>` into an Action method. The singular literally calls `.First()` to get the one file

**Comment by JoeStead** [comment]
We should look at getting integration tests running as part of the CI too

File: src/Extensions/BindExtensions.cs (hunk 2 above)
Comment (JoeStead) [reply src/Extensions/BindExtensions.cs]: Added this (by just returning that matches, no point looping through them all).

Comment (JoeStead) [reply src/Extensions/BindExtensions.cs]: Then again, this causes duplication a bit 🤔

**Comment by jchannon** [comment]
I think have the loop in a method. BindFiles and BindFile calls it. The
latter just calls .First()

**Comment by JoeStead** [comment]
Yeah, didn't want to do that because the latter will loop through all the files and just take the first one. If we had `yield return` inside an async scope, it'd be fine

**Comment by jchannon** [comment]
ReadFormAsync is the hard part. Unless they are uploading a tonne of files, adding it to a list isn't really going to be that painful.

https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/FormFileModelBinder.cs#L53-L132

**Comment by JoeStead** [comment]
Changed for now, can't say I like it though

**Comment by jchannon** [comment]
It does seem there is a potential DDOS attack there

**Comment by JoeStead** [comment]
Yep, fill the request with thousands of "files" so the server has to loop through them all just to pick the first one.

**Comment by jchannon** [comment]
aye

File: src/Extensions/BindExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+        public static async Task<IEnumerable<IFormFile>> BindFiles(this HttpRequest request)
+        {
+            var postedFiles = new List<IFormFile>();
+
+            if (request.HasFormContentType)
+            {
+                var form = await request.ReadFormAsync();
+
+                foreach (var file in form.Files)
```
Comment (jchannon) [inline src/Extensions/BindExtensions.cs]: Could be ugly but could add a method with optional bool flag `private IEnumerable<IFormFile> GetFiles(bool returnOnFirst = false)` that gets called by BindFile with true set

File: src/Extensions/BindExtensions.cs [hunk 4]
```diff
@@ -61,15 +61,26 @@ public static T Bind<T>(this HttpRequest request)
...
+                    }
                 }
             }
 
             return postedFiles;
         }
 
+        public static async Task<IEnumerable<IFormFile>> BindFiles(this HttpRequest request)
+        {
+            return await request.BindFiles(false);
```
Comment (jchannon) [inline src/Extensions/BindExtensions.cs]: set the parameter name? BindFiles(returnOnFirst:false)

Comment (JoeStead) [reply src/Extensions/BindExtensions.cs]: It's a private method with one parameter, really?

**Comment by jchannon** [comment]
yup :)

**Comment by jchannon** [comment]
Looks good. What are your thoughts on a later PR to change the namespaces. So the Bind extensions go under Botwin.Binding etc

**Pull request merged**